followthemoney/property.py

```diff
@@ -1,21 +1,71 @@
 from rdflib import URIRef  # type: ignore
-from banal import is_mapping
-from normality import stringify
+from banal import is_mapping, as_bool
+from typing import TYPE_CHECKING, Any, List, Optional, TypedDict
 
 from followthemoney.exc import InvalidModel
 from followthemoney.types import registry
 from followthemoney.util import gettext, NS, get_entity_id
 
+if TYPE_CHECKING:
+    from followthemoney.schema import Schema
 
-class Property(object):
+
+class ReverseSpec(TypedDict, total=False):
+    name: str
+    label: Optional[str]
+    hidden: Optional[bool]
+
+
+class PropertyDict(TypedDict, total=False):
+    label: Optional[str]
+    description: Optional[str]
+    type: Optional[str]
+    hidden: Optional[bool]
+    matchable: Optional[bool]
+    # stub: Optional[bool]
+    rdf: Optional[str]
+    range: Optional[str]
+
+
+class PropertySpec(PropertyDict):
+    reverse: ReverseSpec
+
+
+class PropertyToDict(PropertyDict, total=False):
+    name: str
+    qname: str
+    reverse: Optional[str]
+    stub: Optional[bool]
+
+
+class Property:
```

Comment:
Might be worth defining a standard here as I've been putting the root type class at the top. Not bothered where it lives, but I do like consistency.